controlplane/kubeadm/api/v1alpha3/kubeadm_control_plane_types_test.go

```diff
@@ -0,0 +1,116 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha3
+
+import (
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	"golang.org/x/net/context"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	cabpkv1 "sigs.k8s.io/cluster-api/bootstrap/kubeadm/api/v1alpha3"
+)
+
+// These tests are written in BDD-style using Ginkgo framework. Refer to
+// http://onsi.github.io/ginkgo to learn more.
+
+var _ = Describe("KubeadmControlPlane", func() {
```

Comment:
Are we going to be switching these tests to use Ginkgo eventually? Just curious. 🙂

Reply:
which tests are you referring to?

Reply:
whoops. Put this comment in the wrong file. I meant to put it in `controllers/controller_test.go`. I noticed that we have the ginkgo and traditional tests in that file.

Reply:
ginkgo is only used (in Cluster API at least) for integration tests and some e2e tests as well, while normal tests are used for unit testing